ff/src/fields/mod.rs

```diff
@@ -188,6 +188,21 @@ pub trait Field:
         }
         res
     }
+
+    /// Exponentiates a field element by a number represented with `u64` limbs,
+    /// via a table containing as many powers of 2 of the given element
+    /// as the ceil of log2 of the exp, starting from the 1st power,
+    /// i.e. pows_2 = [p, p^2, p^4, ..., p^(2^n)] where exp has <= n bits
+    #[inline]
+    fn pow_with_table<S: AsRef<[u64]>>(pows_2: &[Self], exp: S) -> Self {
+        let mut res = Self::one();
+        for (pow, bit) in BitIteratorLE::without_trailing_zeros(exp).enumerate() {
+            if bit {
+                res *= &pows_2[pow];
+            }
+        }
+        res
+    }
```

Comment:
Let's make it return an option, so that it return `None` when the powers are insufficient

```suggestion
    /// Exponentiates a field element `f` by a number represented with `u64` limbs,
    /// using a precomputed table containing as many 
    /// powers of 2 of `f`
    /// as the ceiling of log2 of the exponent `exp`, starting from the 1st power.
    /// That is, `powers_of_2` should equal `&[p, p^2, p^4, ..., p^(2^n)]` 
    /// when `exp` has at most `n` bits.
    ///
    /// This returns `None` when a power is missing from the
    /// table.
    #[inline]
    fn pow_with_table<S: AsRef<[u64]>>(powers_of_2: &[Self], exp: S) -> Option<Self> {
        let mut res = Self::one();
        for (pow, bit) in BitIteratorLE::without_trailing_zeros(exp).enumerate() {
            if bit {
                res *= &powers_of_2.get(&pow)?;
            }
        }
        Some(res)
```